### PR TITLE
Feature: Version Update Checking

### PR DESCRIPTION
**Pull request by UTDZac**
This adds support for the Tracker to check this repository for release updates. It does so by comparing major, minor, and patch version numbers. As such, the version schema moving forward will be `[Major].[Minor].[Patch]` , eg `6.3.0`

image

When a new release goes out, if the `Major` or `Minor` patch number changes, then the user will be alerted that an update is available. This means that we can push hotfix updates regularly using the `Patch` number, without continuing to alert everyone about updates.

This feature offers a **Remind me later** option if the user wants to be reminded later to update. I think one possible flaw with this is using the Quick-load feature will likely trigger the version update reminder, every time. Unsure if there is a good way to go about this to differentiate between reloading the script from a quick-load and reloading the script on a different day.

### CODE REVIEW

**Comment by UTDZac** [comment]
Two updates here:
- Version Checking is only supported in Windows, so there is now a check for that before attempting.
- An update notification will only appear once per real life day, for cases where quickload reloads the Tracker script

**Comment by UTDZac** [comment]
Just leaving a note that one use-case here that's hard to test live, but I think I thought it through correctly, is the following:

1. User currently on v6.3.0 is notified of a new release: v6.4.0
2. User dismisses the version update and purposefully does not update
3. Sometime in the future a new release yet again comes out: v6.5.0
4. At this point, the user should again be notified about a new version update

**Review by Fellshadow — COMMENTED** [review]
I'm happy enough for this to get merged in, just one or two thoughts I had that i'd like addressed first

Will be super cool to have this in next release though, for those who don't keep as close an eye on the discord / github pages

File: Ironmon-Tracker.lua [hunk 1]
```diff
@@ -194,6 +204,78 @@ function Main.Run()
...
+	local output = pipe:read("*all") or ""
+	local _, _, major, minor, patch = string.match(output, '"tag_name":(%s+)"(%w+)(%d+)%.(%d+)%.(%d+)"')
+
+	local dayOfMonth = os.date("%d")
+	local latestVersion = major .. "." .. minor .. "." .. patch
+	local newVersionAvailable = Main.Version.major ~= major or Main.Version.minor ~= minor
+
+	-- Only notify about updates once per day
+	local shouldNotify
+	if dayOfMonth == Main.Version.daychecked then
```
[the remark this answers is not in the document]
Comment (UTDZac) [reply Ironmon-Tracker.lua]: Good note, I've made this change in the latest commit.

**Review by Fellshadow — APPROVED** [review]

**Pull request merged**